eva-stats/stats_variant_warehouse.py

```diff
@@ -0,0 +1,181 @@
+    database_name = 'eva_{0}_{1}'.format(result[0][0], result[0][1])
+    return database_name
+
+
+def get_mongo_primary_host_and_port(mongo_hosts_and_ports):
+    """
+    :param mongo_hosts_and_ports: All host and ports stored in the private settings xml
+    :return: mongo primary host and port
+    """
+    for host_and_port in mongo_hosts_and_ports.split(','):
+        if '001' in host_and_port:
+            properties = host_and_port.split(':')
+            return properties[0], properties[1]
+
+
+def get_from_variant_warehouse(mongo_handle, metadata_handle, projects):
+    projects_evapro = "('" + projects.replace(",", "','") + "')"
+    query = "select a.vcf_reference_accession, p.project_accession, a.analysis_accession, af.file_id, " \
+            "f.filename, f.file_type, asm.taxonomy_id " \
+            "from analysis a " \
+            "left join analysis_file af on (af.analysis_accession = a.analysis_accession)" \
+            "left join file f on (f.file_id = af.file_id)" \
+            "left join project_analysis p on (p.analysis_accession = a.analysis_accession)" \
+            "left join assembly asm on (asm.assembly_accession = a.vcf_reference_accession)" \
+            "where f.file_type = 'VCF' " \
+            "and p.project_accession in {0}" \
+            "order by a.vcf_reference_accession, p.project_accession, a.analysis_accession".format(projects_evapro)
+    logger.info("Querying EVAPRO")
+    result_evapro = get_all_results_for_query(metadata_handle, query)
+
+    analyses = []
+    analysis_filename = {}
+    for index, row in enumerate(result_evapro):
+        insert_into_stats(metadata_handle, row)
+        analysis = row[2]
+        filename = row[4]
+        analyses.append(analysis)
+        analysis_filename[analysis] = filename
+        # All except last row
+        if index < len(result_evapro) - 1:
+            next_row = result_evapro[index + 1]
+            if row[1] != next_row[1]:
+                # if next row is a new project
+                assembly = row[0]
+                project = row[1]
+                get_counts_from_variant_warehouse(assembly, project, analyses, metadata_handle, mongo_handle)
+                get_dates_from_variant_warehouse(assembly, project, analysis_filename, metadata_handle, mongo_handle)
+                analyses = []
+                analysis_filename = {}
+        else:
+            assembly = row[0]
+            project = row[1]
+            get_counts_from_variant_warehouse(assembly, project, analyses, metadata_handle, mongo_handle)
+            get_dates_from_variant_warehouse(assembly, project, analysis_filename, metadata_handle, mongo_handle)
+
+
+def insert_into_stats(metadata_handle, row):
+    assembly, project, analysis, file_id, filename, file_type, taxonomy_id = row[0], row[1], row[2], row[3], row[4], \
+                                                                             row[5], row[6]
```

Comment:
Have you tried just this?
```suggestion
assembly, project, analysis, file_id, filename, file_type, taxonomy_id = row
```

Reply:
Thanks, changed